### PR TITLE
Make differenceInBusinessDays return NaN instead of Invalid Date

### PR DESCRIPTION
**Pull request by dmarcr1997**
Saw there were already tests in place for this issue. I got them to pass by changing the return from new Date(NaN), which returns invalid date, to NaN.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "date-fns",
+  "version": "DON'T CHANGE; IT'S SET AUTOMATICALLY DURING DEPLOYMENT; ALSO, USE YARN FOR DEVELOPMENT",
+  "lockfileVersion": 1
+}
```
Comment (kossnocorp) [inline package-lock.json:5]: Please remove the file from the PR.

File: src/differenceInBusinessDays/index.js [hunk 2]
```diff
@@ -41,7 +41,9 @@ export default function differenceInBusinessDays(
   var dateLeft = toDate(dirtyDateLeft)
   var dateRight = toDate(dirtyDateRight)
 
-  if (!isValid(dateLeft) || !isValid(dateRight)) return new Date(NaN)
+  // fix for issue # 2012
+  // returning undefined to fix issue of returning a Date when either date is invalid
+  if (!isValid(dateLeft) || !isValid(dateRight)) return NaN
```
Comment (kossnocorp) [inline src/differenceInBusinessDays/index.js:46]: No need for the comment.

```suggestion
  if (!isValid(dateLeft) || !isValid(dateRight)) return NaN
```

**Review by kossnocorp — CHANGES_REQUESTED** [review]
Thank you for the change, but I will not merge it right now as it introduces a breaking change. However, it will be handy when I start working on v3. Tests actually check for Invalid Date, which internally represented by NaN, so the PR didn't fix tests 😉

**Comment by fturmel** [comment]
Going to close this PR if you don't mind. This got fixed and merged when the function was converted to TypeScript (perhaps early by accident), and it's been in v2 releases for a while. See #2012 and #2414

**Pull request closed without merging**